packages/payment/src/services/payment-module.ts

```diff
@@ -207,38 +207,33 @@ export default class PaymentModuleService<
     data: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
   ): Promise<PaymentSessionDTO> {
-    const created = await this.paymentSessionService_.create(
-      {
+    const { result, errors } = await paymentSessionWorkflow(
+      this.__container__ as any
+    ).run({
+      input: {
+        payment_collection_id: paymentCollectionId,
         provider_id: data.provider_id,
-        amount: data.providerContext.amount,
-        currency_code: data.providerContext.currency_code,
-        payment_collection: paymentCollectionId,
+        amount: data.amount,
+        currency_code: data.currency_code,
+        context: data.context,
+        data: data.data,
       },
-      sharedContext
-    )
+      context: sharedContext,
+    })
 
-    try {
-      const sessionData = await this.paymentProviderService_.createSession(
-        data.provider_id,
-        {
-          ...data.providerContext,
-          resource_id: created.id,
-        }
-      )
+    if (errors.length) {
+      throw errors[0].error
+    }
```

Comment:
comment: Might make sense to come up with a pattern for handling errors from workflows within services

Reply:
Maybe instead of having a throwOnError configuration, we could have an onError callback. If it is provided you receive all the errors and can process them as you want and if not provided then it will throw. Then we can have default onError handler? 🤔